diff -Naur gnupg-2.1.13/g10/keyserver.c gnupg-2.1.13-patched/g10/keyserver.c
--- gnupg-2.1.13/g10/keyserver.c	2016-06-16 08:23:13.000000000 -0700
+++ gnupg-2.1.13-patched/g10/keyserver.c	2016-06-22 21:40:54.353265999 -0700
@@ -127,12 +127,10 @@
     *p = 0;
 
   if (!strcmp (option, "ca-cert-file"))
-    log_info ("keyserver option '%s' is obsolete; please use "
-              "'%s' in dirmngr.conf\n",
-              "ca-cert-file", "hkp-cacert");
+      ;
   else if (!strcmp (option, "check-cert")
            || !strcmp (option, "broken-http-proxy"))
-    log_info ("keyserver option '%s' is obsolete\n", option);
+      ;
   else if (noisy || opt.verbose)
     log_info ("keyserver option '%s' is unknown\n", option);
 }
